cm: storage state interfaces

mlohvynenko opened the pull request.

mlohvynenko committed:
cm: storagestate: add storage state interfaces

This patch adds storage state interfaces to the cm module.

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
common: cloudprotocol: add state structures

This patch adds cloudprotocol state structures.

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
common: types: add less operator for instance ident

This patch makes it possible to use instance ident
with STL containers that require a less operator.

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
common: crypto: support SHA3-224 hash algorithm

This patch adds support for the SHA3-224 hash algorithm
in the crypto provider.

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
common: tools: enhance fs platform interface with set user quota

This patch enhances the platform filesystem interface
by adding a method to set user quota.

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
common: config: remove redundant sha1 digest size definition

This patch removes the redundant definition of the SHA1 digest
size in the configuration header file.

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img commented on include/aos/common/cloudprotocol/state.hpp:
```diff
@@ -0,0 +1,169 @@
+ */
+static constexpr auto cStateReason = cErrorMessageLen;
+
+/**
```
It could be one enum for all cloud messages I guess. Do you need this enum right now?

al1img commented on include/aos/cm/storagestate/storagestate.hpp:
```diff
@@ -0,0 +1,96 @@
+     * @param instanceIdent instance ident.
+     * @return Error.
+     */
+    virtual Error RemoveServiceInstance(const InstanceIdent& instanceIdent) = 0;
```
Let's rename to just `Remove`

al1img commented on include/aos/cm/storagestate/storagestate.hpp:
```diff
@@ -0,0 +1,96 @@
+     * @param state state to accept.
+     * @return Error.
+     */
+    virtual Error StateAcceptance(const cloudprotocol::StateAcceptance& state) = 0;
```
AcceptState ?

mlohvynenko replied:
Agree. Your proposal looks better to me as well
But current name is consistent with aos protocol

al1img commented on src/common/crypto/mbedtls/cryptoprovider.cpp:
```diff
@@ -608,6 +608,8 @@ RetWithError<UniquePtr<HashItf>> MbedTLSCryptoProvider::CreateHash(Hash algorith
 
     if (algorithm.GetValue() == HashEnum::eSHA256) {
         alg = PSA_ALG_SHA_256;
+    } else if (algorithm.GetValue() == HashEnum::eSHA3_224) {
```
What about openssl crypto provider?

mykola-kobets-epam commented on include/aos/cm/storagestate/storagestate.hpp:
```diff
@@ -0,0 +1,182 @@
+};
+
+/**
+ * Storage state interface.
```
not informative. Suggestion: interface to manage storage & state partitions.

mykola-kobets-epam reviewed (approved):
`Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>`

mlohvynenko committed:
cm: communication: add communication interface

This patch adds a new communication interface to the cm module
with API needed for storage state module.

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img reviewed (approved):
`Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>`

MykolaSuperman reviewed (approved):
`Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>`
